File: ios/AppGroupModule.m
//
//  AppGroupModule.m
//  Voicenotes
//
//  Created by Andriy Hrytsyshyn on 7/5/24.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(AppGroupModule, NSObject)

RCT_EXTERN_METHOD(setValueInAppGroup:(NSString *)key value:(NSString *)value)

@end

@interface RCT_EXTERN_MODULE(ActionModule, RCTEventEmitter)

RCT_EXTERN_METHOD(startRecord)
RCT_EXTERN_METHOD(askAI)
RCT_EXTERN_METHOD(searchNote)
RCT_EXTERN_METHOD(sendToken)

@end

File: ios/AppGroupModule.swift
//
//  AppGroupModule.swift
//  Voicenotes
//
//  Created by Andriy Hrytsyshyn on 7/5/24.
//

// AppGroupModule.swift
import Foundation

@objc(AppGroupModule)
class AppGroupModule: NSObject {
  
  @objc func setValueInAppGroup(_ key: String, value: String) {
    let sharedUserDefaults = UserDefaults(suiteName: "group.watchOS.storage")
    sharedUserDefaults?.set(value, forKey: key)
    print("successfully save to user defaulr")
  }
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
